packages/amplication-git-utils/src/git/github.service.ts

```diff
@@ -41,25 +48,154 @@ export class GithubService implements GitProvider {
   private privateKey: string;
   private gitInstallationUrl: string;
   private octokit: Octokit;
-
-  constructor(private readonly gitProviderArgs: GitProviderArgs) {
+  public readonly name = EnumGitProvider.Github;
+  public readonly domain = "github.com";
+  constructor(private readonly gitProviderArgs: GitProviderConstructorArgs) {
     const {
       GITHUB_APP_INSTALLATION_URL,
       GITHUB_APP_APP_ID,
       GITHUB_APP_PRIVATE_KEY,
     } = process.env;
+    if (!GITHUB_APP_INSTALLATION_URL) {
+      throw new Error("GITHUB_APP_INSTALLATION_URL is not defined");
+    }
     this.gitInstallationUrl = GITHUB_APP_INSTALLATION_URL;
+    if (!GITHUB_APP_APP_ID) {
+      throw new Error("GITHUB_APP_APP_ID is not defined");
+    }
     this.appId = GITHUB_APP_APP_ID;
+    if (!GITHUB_APP_PRIVATE_KEY) {
+      throw new Error("GITHUB_APP_PRIVATE_KEY is not defined");
+    }
     this.privateKey = GITHUB_APP_PRIVATE_KEY;
 
     const privateKey = this.getFormattedPrivateKey(this.privateKey);
-
     this.app = new App({
       appId: this.appId,
       privateKey,
     });
   }
 
+  getCloneUrl({ owner, repositoryName }: CloneUrlArgs) {
+    return `https://${this.domain}/${owner}/${repositoryName}.git`;
+  }
+
+  async getCurrentUserCommitList(args: GetBranchArgs): Promise<Commit[]> {
+    const { branchName, owner, repositoryName } = args;
+    const currentUserData = await this.getCurrentUser();
+
+    let moreCommitsPagination = true;
+    let commitsList: Commit[] = [];
+    let cursor: string | undefined = undefined;
+
+    do {
+      const { commits, hasNextPage, endCursor } =
+        await this.paginatedCommitsList({
+          botData: currentUserData,
+          branchName,
+          owner,
+          repositoryName,
+          cursor,
+        });
+      moreCommitsPagination = hasNextPage;
+      commitsList = commitsList.concat(commits); // The list is in ascending order ( newest commits are first )
+      cursor = endCursor;
+    } while (moreCommitsPagination);
+
+    return commitsList;
+  }
+
+  private async paginatedCommitsList(
+    args: GetBranchArgs & {
+      cursor: string | undefined;
+      botData: GitUser;
+    }
+  ): Promise<{ commits: Commit[]; hasNextPage: boolean; endCursor: string }> {
+    const { branchName, owner, repositoryName, cursor, botData } = args;
+
```

Comment:
It is usually better to define a variable with the pagination size for code readability.